extractor/service.go

```diff
@@ -229,16 +260,30 @@ func findAllFetchedReposWithRef(refText string) map[model.SHA1]string {
 				break
 			}
 		}
-		if masterRef == nil { // skipping repos \wo it
+		if masterRef == nil {
+			// skipping repos \wo it
 			log.Warn("No reference found ", "repo", repo.ID, "reference", refText)
 			continue
 		}
 
-		repos[masterRef.Init] = repo.ID.String()
+		repos[masterRef.Init] = repo
 	}
 	return repos
 }
 
+// If N=0, get total number of fetched respoitories from DB. Use N otherwise.
+func allOrN(n uint64) uint64 {
```

Comment:
This may be a bit miss-leading, but the reason for this code is - a convenience for debugging. I found that it's very convenient to know exact `N` and print it to the logs (before iterating over anything).

That is why there is query to DB here, in case when user did not provide us with desired limit.